### PR TITLE
bpf: redirect fixes and follow-ups

### PR DESCRIPTION
**Pull request by borkmann**
see commit msgs

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: daemon/cmd/status.go [hunk 1]
```diff
@@ -161,6 +161,16 @@ func (d *Daemon) getBandwidthManagerStatus() *models.BandwidthManager {
 	return s
 }
 
+func (d *Daemon) getHostRoutingStatus() *models.HostRouting {
+	s := &models.HostRouting{}
+	if option.Config.EnableHostLegacyRouting {
+		s.Mode = models.HostRoutingModeLegacy
+	} else {
+		s.Mode = models.HostRoutingModeBPF
+	}
```
Comment (tklauser) [inline daemon/cmd/status.go]: Nit: could be simplified to
```suggestion
	s := &models.HostRouting{Mode: models.HostRoutingModeBPF}
	if option.Config.EnableHostLegacyRouting {
		s.Mode = models.HostRoutingModeLegacy
	}
```

**Comment by borkmann** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]

File: daemon/cmd/kube_proxy_replacement.go [hunk 2]
```diff
@@ -258,7 +258,6 @@ func initKubeProxyReplacementOptions() (strict bool) {
 				_, found = h["bpf_redirect_neigh"]
 			}
 			if !found {
-				log.Warnf("Re-enabling host legacy routing as bypassing needs kernel 5.10.0 or newer.")
```
Comment (brb) [inline daemon/cmd/kube_proxy_replacement.go:261]: Maybe still an info msg for users to be able to determine an exact datapath configuration from a log?

File: bpf/lib/common.h [hunk 3]
```diff
@@ -800,15 +801,16 @@ struct lb6_src_range_key {
 	union v6addr addr;
 };
 
-static __always_inline int redirect_peer(int ifindex __maybe_unused,
-					 __u32 flags __maybe_unused)
+static __always_inline int redirect_ep(int ifindex __maybe_unused,
+				       bool needs_backlog __maybe_unused)
```
Comment (brb) [inline bpf/lib/common.h:805]: Maybe short comment for future code readers about `needs_backlog` meaning?

**Review by brb — COMMENTED** [review]
LGTM, just two nits.

File: daemon/cmd/kube_proxy_replacement.go (hunk 2 above)
Comment (borkmann) [reply daemon/cmd/kube_proxy_replacement.go:261]: I added cilium status dump which might be easier to introspect on a live system than old log potentially, given we also don't add any Info msg when pre-conditions don't apply, I removed it from here. We can definitely revisit though.

**Comment by borkmann** [comment]
(Prior CI run all green, added comment to BPF C code only.)

**Pull request merged**